### ISSUE TITLE
Next Sched Date Advances Despite Failed Transaction

### ISSUE THREAD

**Issue by shaigluskin** [iATSPayments/com.iatspayments.civicrm#27]
I almost posted this as a comment in #24, but I think it is a different issue.

Alan wrote in #24

> When a recurring transaction fails, it will keep getting tried until it goes through.

That's not my experience. Well, let me qualify. That _is_ my experience if the cause of the failed transaction is an expired credit card. Note that in the case of an expired credit card, iatspayments.com never gets touched. The rejection happens prior to anything being sent to iatspayments.com

However, if there is a different reason for the rejection, which is determined at IATS servers, the next date in that case gets pushed one month forward from the time of the transaction. 

In my case in the last two days, 118 transactions were processed and there were 12 rejects of type code 15. Those were all caught by the IATS server and in every case the next sched date was pushed up by a month.

The status was appropriately set to "Failed", but the next sched date was advanced.

**Comment by sgladstone** [comment]
The code in core for Authorize.net allows for the back-office CiviCRM user to change the card details (ie the expiration date, the billing address and/or the card number.)  On the contribution tab for a contact, there is an "edit" link next to each recurring subscription record (i.e. at the bottom of the tab) its alongside the link to cancel the entire schedule. 

Perhaps the scheduled job can retry only if the user has edited something. ( ie there is a reasonable chance of success since something been updated.) (On a related note, there should be a way to link a manual one-off contribution to the schedule, such as when a donor/member has mailed a check to cover the failed transaction)

**Comment by adixon** [comment]
Yes, that's on our road map for the next release, more sophisticated handling of failures, thanks for the notes.
